### PR TITLE
Model additions

### PR DESCRIPTION
**Pull request by dvanetten03**
Can merge now. More updates to come.

### CODE REVIEW

**Comment by dvanetten03** [comment]
This is just updating all models so we can see where we are at, what else we need and any refactoring or moving that needs done!

**Comment by ghost** [comment]
Hey Diane, Anything specific you want help with? I'll take a look at it now before we meet up tonight.

**Comment by dvanetten03** [comment]
@davidslee101 I was just putting this out here, along with the google doc that I sent so we could see what we needed and what was out there.  It may/may not be repeat of something else we've done, but I was trying to get everything down so we could get the methods done and move on.

**Comment by ghost** [comment]
yes, that sounds good. I think we need this logic to figure out how to
capture a piece, so we should just comment it out of the queen.rb.

However, it still bugs me that I can't figure out why it's causing errors.
In my head there nothing wrong with the code as is.

**Comment by ghost** [comment]
Hey Diane, I have an answer that's not an answer.
You have `def piece_exists` in queen.rb and in piece. rb
I think we only need it in piece.rb, so we could probably remove this method from queen.rb. Once you do this, all the tests pass.

The other thing is that we have the `is_obstructed?` method from Anna's PR, so we could probably do without the piece_exists method, unless you were planning to turn this into part of the capturing a piece method.

**Comment by dvanetten03** [comment]
My concerns were not about the processes, if you read my comments, but Anna put the comments in the middle of her PR where it has no meaning. I understand she was looking at my code, but she commented on hers and that's my issue. I knew there were redundancies, if you read the Google doc that had everything in it you could see that. I was trying to make it easy for everyone to see and for everyone to be able to refactor with ease. We are all learning to program, that's why we are doing this project, at least that's the assumption I'm operating under. So I am happy to get/give feedback, but it's important to put that feedback within the code where the issue is. Moving forward, as we work at companies with big projects it's important to be specific and point out exactly where an issue is because there could be 50,000 lines of code and 20 different PRs that someone has submitted so they won't know what you're talking about if you put the info in your own PR. 

Diane

File: app/models/knight.rb [hunk 1]
```diff
@@ -22,4 +22,16 @@ def legal_move?(x, y)
...
+
+  def valid_move?(row, col)
+  end
+
+  def piece_where_moving_to?(row, col)
+  end
+
+  def piece_still_on_board?(row, col)
+  end
+
```
Comment (washburnad) [inline app/models/knight.rb:36]: These 3 methods should be identical for all pieces so can be moved up to the superclass.

File: app/models/piece.rb [hunk 2]
```diff
@@ -8,77 +8,87 @@ class Piece < ActiveRecord::Base
...
+    (self.row_position - row_position).abs == (self.col_position - col_position).abs
+  end
 
-def legal_diag_move?(row, col)
-  #row >= 0 && row <= 7 && col >= 0 && col <= 7
-  (self.row_position - row_position).abs == (self.col_position - col_position).abs
-end
+  def piece_exists?(row, col)
+    Piece.where(game_id: self.id, row_position: row, col_position: col).exists?
+  end
```
Comment (washburnad) [inline app/models/piece.rb:81]: use pieces.where( ) and don't add game_id in the params.  Rails will take care of the association.

**Comment by ghost** [comment]
nice! all the pieces now have logic to determine legal moves. I give it a thumbs up!

One question I have is on the diagonal move method in piece.rb
  `def legal_diag_move?(row, col)
    #abs val row must equal abs value col
    (self.row_position - row_position).abs == (self.col_position - col_position).abs
  end`

when you have self.row_position, is this the final destination row or the starting point row? Still a little hazy on the self.row_position stuff.

**Pull request closed without merging**